### PR TITLE
added saucerswap faucet bot code

### PR DESCRIPTION
**Pull request by vea-vecturne**
**Description**:
This PR adds the SaucerSwap Labs faucet bot repo in it's current state to this repository.

### CODE REVIEW

File: saucerswap-faucetbot/package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "discordbots",
```
Comment (web3-nomad) [inline saucerswap-faucetbot/package.json:2]: name should match directory

File: saucerswap-faucetbot/main.js [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+  } catch (e) {
+    console.debug("Failed to retrieve user message!");
+    return;
+  }
+  const time = getTimeTilReset();
+
+  // switch-case on commands
+  switch (cmd) {
+    case "!manual":
+      const response = `Faucet Bot is here to dispense *$SAUCE!*
```
Comment (web3-nomad) [inline saucerswap-faucetbot/main.js:81]: make this a configuration option.. for your own sake.. people will get confused if there is SAUCE offered everywhere

File: saucerswap-faucetbot/main.js [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+          +"\n3. Use the HBAR you got from Step 2 to fuel your journey on SaucerSwap!\n"
+          +"\nThe HBAR you received will pay for the association fee, network fee, and gas fee incurred by the process outlined above.\n"
+          +"\nWelcome to SaucerSwap! <:SAUCE:946907336742682624>", userMsg);
+          accountIdsBeingProcessed.delete(accountId);
+        } else {
+          throw new Error(`Tx receipt status is not a success!`);
+        }
+      } catch (e) {
+        replyThenDelete(
+          `HBARs couldn't be dispensed! <@453749295309389834> HEEEEELP!!! <:aaa:921863627655508008>`,
```
Comment (web3-nomad) [inline saucerswap-faucetbot/main.js:193]: make this a conifg

File: saucerswap-faucetbot/main.js [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+        // if tx success, insert values into table...
+        if (receipt.status._code == 22) {
+          await db.query( `INSERT INTO faucet_tracker (accountId) VALUE (?);`, [accountId] );
+          console.debug( `${payout / Math.pow(10, 6)} $SAUCE sent to ${accountId}` );
+          replyThenDelete(`Success!`, userMsg);
+        } else {
+          throw new Error(`Tx receipt status is not a success!`);
+        }
+      } catch (e) {
+        replyThenDelete(`Something went wrong! <@453749295309389834> HEEEEELP!!! <:aaa:921863627655508008>`, userMsg);
```
Comment (web3-nomad) [inline saucerswap-faucetbot/main.js:337]: user-config

**Review by web3-nomad — CHANGES_REQUESTED** [review]

File: saucerswap-faucetbot/package.json (hunk 1 above)
Comment (hhe44) [reply saucerswap-faucetbot/package.json]: Resolved in the latest commit.

File: saucerswap-faucetbot/main.js (hunk 2 above)
Comment (hhe44) [reply saucerswap-faucetbot/main.js]: Resolved in the latest commit.

File: saucerswap-faucetbot/main.js (hunk 3 above)
Comment (hhe44) [reply saucerswap-faucetbot/main.js]: image

That last part of the string is used to alert a developer via discord if the bot fails for whatever reason. I'm removing it if other devs want to customize it to their liking.

File: saucerswap-faucetbot/main.js (hunk 4 above)
Comment (hhe44) [reply saucerswap-faucetbot/main.js]: image

That last part of the string is used to alert a developer via discord if the bot fails for whatever reason. I'm removing it if other devs want to customize it to their liking.

**Review by web3-nomad — APPROVED** [review]
LGTM

**Review by web3-nomad — DISMISSED** [review]

**Comment by hhe44** [comment]
Great! Although the merge is currently blocked:
image

Can you mark the first conversation as resolved?
image

Should be able to pass all checks after that

**Review by hhe44 — DISMISSED** [review]

**Comment by hhe44** [comment]
Will need approval to merge @web3-nomad

**Review by web3-nomad — APPROVED** [review]

**Comment by hhe44** [comment]
Encountering an odd issue trying to merge
image

Not entirely sure why it's still being blocked
image

**Comment by web3-nomad** [comment]
GPG sign. 'capital S on the commit, not lowercase'

**Comment by hhe44** [comment]
Requesting approval again @web3-nomad

**Review by web3-nomad — APPROVED** [review]

**Pull request merged**